src/TransformationUtils.cc

```diff
@@ -134,6 +134,7 @@ std::vector<EId> detectLoop(const ChcDirectedGraph & graph) {
     std::unordered_set<SymRef, SymRefHash> visitedVertices;
     std::unordered_set<SymRef, SymRefHash> verticesOnStack;
     visit(visitedVertices, verticesOnStack, graphRepresentation, graph, graph.getEntry(), loop);
+    reverse(loop.begin(), loop.end());
```

Comment:
Why do you need to reverse the loop now?

Reply:
Also, should be `std::reverse`, no?

Reply:
Main reason is I want to have a representation of the loop in which it is present in the current active path, but before reverse it is in the reversed order

Reply:
OK. Just make it `std::reverse`.

Reply:
Fixed